embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -228,21 +236,46 @@ private FileList listFiles(PluginTask task)
             }
             throw ex;
         }
+        catch (InterruptedException | RetryGiveupException ex) {
+            throw new AmazonServiceException(ex.getMessage());
```

Comment:
I don't think wrap up our exception with a library exception is a good idea, it is prone for falsehood assumptions (this exception doesn't have requestId, errorCode, statusCode,...). Also, this will lost the original cause with stacktrace. If we're trying to re-throw an unchecked exception here, I think a simple `RuntimeException(ex)` is sufficient and will retain all the original information.